imagegw/shifter_imagegw/api.py

```diff
@@ -211,6 +216,79 @@ def pull(system, imgtype, tag):
     return jsonify(create_response(rec))
 
 
+# Import image
+# This will import the requested image from a file path on the system.
+@app.route('/api/doimport/<system>/<imgtype>/<path:tag>/', methods=["POST"])
+def doimport(system, imgtype, tag):
+    """ 
+    Pull a specific image and tag for a systems. 
+    """
+    if imgtype == "docker" and tag.find(':') == -1:
+        tag = '%s:latest' % (tag)
+
+    auth = request.headers.get(AUTH_HEADER)
+    app.logger.debug("inside doimport")
+    data = {}
+    try:
+        rqd = request.get_data()
+        app.logger.debug("rqd '%s'" % (rqd))
+        if rqd is not "" and rqd is not None:
+            data = json.loads(rqd)
+            app.logger.debug("data '%s'" % (data))
+    except:
+        app.logger.warn("Unable to parse doimport data '%s'" %
+                        (request.get_data()))
+        pass
+
+    memo = "import system=%s imgtype=%s tag=%s" % (system, imgtype, tag)
+    app.logger.debug(memo)
+    i = {'system': system, 'itype': imgtype, 'tag': tag}
```

Comment:
It occurs to me that one user could overwrite another users tag.  For now this may be okay since this would be a limited/admin like function.  But this is a place where ACLs could be used.

Reply:
Yeah, this is a good point. As a first pass, I'd think making these images only overwriteable by the group members of the user who uploaded them would be good. But I'm not sure if we'd be able to pick up the correct group from the uploaded. Maybe we discuss this offline?